### PR TITLE
i.sentinel.import enhancements

### PR DESCRIPTION
**Pull request by ninsbl**
Various enhancement:
 - Do not unzip if already done so (speed improvement)
 - Add filename pattern option (speed improvement if not all data is supposed to be imported)
 - Remove hardcoded version for Band reference (replace == 7.9.dev with >= 7.9)
 - Import also SCL data for L2A products (could be used as cloud / quality mask)
 - Add o-flag for overriding projection check (for insignificant CRS mismatch (e.g. different EPSG codes for identical CRS))
 - Write metadata into supportfile
 - add unzip_dir option, so users can choose where to unzip to (e.g. if they do not have write access to input dir or want to store SAFE files at a different place e.g. on SSD...)

### CODE REVIEW

**Comment by ninsbl** [comment]
Another possible improvement would be to write more metadata into support file, e.g.:

 - [ ] azimuth
 - [ ] zenith
 - [ ] timestamp
 - [ ] more?, quality indicators? (suggestions are welcome)

Metadata should be written in a machine readable way (e.g. azimuth=15.345, one at a line), so it could be used in subsequent modules (e.g. i.sentinel.mask, i.sentinel.preproc, i.topo.corr) without going back to the original metadata file...

**Comment by ninsbl** [comment]
Might be an advantage to have a new "unzip_dir" option, so users can choose where data should be written, cause the do not necessarily have write access to the directory where Sentinel files are stored...
Opinions?

**Comment by ninsbl** [comment]
I would like to add the unzip_dir option mentioned above.
Also, it does not seem suitable to write metadata into description (as multiple lines don`t seem to be supported). Will append them to history instead...

**Comment by ninsbl** [comment]
Now it should be ready for review

**Comment by veroandreo** [comment]
+1 for these enhancements!

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 1]
```diff
@@ -119,6 +177,28 @@ <h3>Import Sentinel data</h3>
...
+<div class="code"><pre>
+i.sentinel.import input=data pattern_file="MSIL2A.*T32VNR_2019" pattern='B(03|04)'
+</pre></div>
+
+<p>
+Limit import to only bands 3 and 4 from level 2A products for tile T32VNR in 2019,
+unzip to directory "safefiles"
+
+<div class="code"><pre>
+i.sentinel.import input=data unnzip_dir=safefiles pattern_file="MSIL2A.*T32VNR_2019" pattern='B(03|04)'
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
i.sentinel.import input=data unzip_dir=safefiles_dir pattern_file="MSIL2A.*T32VNR_2019" pattern='B(03|04)'
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.py [hunk 2]
```diff
@@ -26,8 +26,18 @@
 #% description: Name of input directory with downloaded Sentinel data
 #% required: yes
 #%end
+#%option G_OPT_M_DIR
+#% key: unzip_dir
+#% description: Name of directory into which zip-files are extracted (Default=input)
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.py]:
```suggestion
#% description: Name of directory into which Sentinel zip-files are extracted (default=input)
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.py [hunk 3]
```diff
@@ -56,6 +66,11 @@
 #% guisection: Settings
 #%end
 #%flag
+#% key: o
+#% description: Override projection check
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.py]:
```suggestion
#% description: Override projection check (use current location's projection)
```
... changed to standard wording (see e.g. r.import)

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 4]
```diff
@@ -6,12 +6,22 @@ <h2>DESCRIPTION</h2>
...
 Note that in the case that spatial reference system of input data differs
-from GRASS GIS location, the input data are reprojected. To speed up this
+from GRASS GIS location, the input data need to be reprojected with 
+<em><a href="r.import.html">r.import</a></em>. To speed up this
 process, a higher than default value can be specified for the <b>memory</b>
 option.
 
+<p>
+In order to ignore insignificant mismatch of the spatial reference 
+systems, reprojection can be supressed with the <b>-o</b> flag.
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
systems, the projection check can be suppressed with the <b>-o</b> flag.
```
... as -o does that.

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 5]
```diff
@@ -47,6 +57,54 @@ <h2>DESCRIPTION</h2>
 </tbody>
 </table>
 
+<p>
+Level 2A products for Sentinel-2 come with a Scene classification (SCL layer)
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
Level 2A products for Sentinel-2 come with a scene classification (SCL layer)
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 6]
```diff
@@ -47,6 +57,54 @@ <h2>DESCRIPTION</h2>
...
+with 20m and 60m resolution, that e.g. can be used for masking clouds and snow
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
at 20m and 60m resolution, that e.g. can be used for masking clouds and snow
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 7]
```diff
@@ -47,6 +57,54 @@ <h2>DESCRIPTION</h2>
...
+and is also imported by default.
+
+<p>
+For each imported band both scene and band specific metadata on geometric
+conditions as well as quality indicators are written into the map history
+(<em><a href="r.support.html">r.support</a></em>). In addition, the scene
+name is stored as <em>source1</em> and the imported or linked file name as
+<em>source2</em>. Also, sensing time is written into the timestamp of the
+map. After import, the metadata below can be retrieved with
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
map. After import, the metadata can be retrieved with <em>r.info -e</em>
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 8]
```diff
@@ -47,6 +57,54 @@ <h2>DESCRIPTION</h2>
...
+<em>r.info -e</em>.
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
as shown below.
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 9]
```diff
@@ -119,6 +177,28 @@ <h3>Import Sentinel data</h3>
...
 
+<p>
+Link data from specific UTM zone while ignoring projection check
+
+<div class="code"><pre>
+i.sentinel.import -l -o input=data pattern_file="_T32"
+</pre></div>
+
+<p>
+Limit import to only bands 3 and 4 from level 2A products for tile T32VNR in 2019
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
Limit import to only bands 3 and 4 from level 2A products for tile T32VNR in 2019:
```

File: grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html [hunk 10]
```diff
@@ -119,6 +177,28 @@ <h3>Import Sentinel data</h3>
...
+
+<div class="code"><pre>
+i.sentinel.import input=data pattern_file="MSIL2A.*T32VNR_2019" pattern='B(03|04)'
+</pre></div>
+
+<p>
+Limit import to only bands 3 and 4 from level 2A products for tile T32VNR in 2019,
+unzip to directory "safefiles"
```
Comment (neteler) [inline grass7/imagery/i.sentinel/i.sentinel.import/i.sentinel.import.html]:
```suggestion
unzip to directory "safefiles_dir":
```

**Pull request merged**